core/lib/spree/app_configuration.rb

```diff
@@ -169,6 +169,10 @@ class AppConfiguration < Preferences::Configuration
     #   @return [String] Email address used as +From:+ field in transactional emails.
     preference :mails_from, :string, default: 'solidus@example.com'
 
+    # @!attribute [rw] associate_user_in_authentication_extension
+    #   @return [Boolean] manually call associate_user instead of relying on current_order
+    preference :associate_user_in_authentication_extension, :boolean, default: false
```

Comment:
We can leverage [the new system for defaults by version](https://github.com/solidusio/solidus/blob/master/CHANGELOG.md#major-changes):

```ruby
preference :associate_user_in_authentication_extension, :boolean, default: by_version(false, "3.2.0.alpha" => true)
```